src/agnostic/gameState.ts

```diff
@@ -59,6 +67,9 @@ export function canAdvance(gameState: GameState, gameEvent: GameEvent): boolean
 	} else if (gameEvent.type === 'inc-player-score') {
 		// can only increase the score of players who exist
 		return !!gameState.players[gameEvent.data.id];
+	} else if (gameEvent.type === 'change-player-name') {
+		// can only change name of players who exist
+		return !!gameState.players[gameEvent.data.id];
```

Comment:
add an `if` check for the new event in the `canAdvance` function. the role of the `canAdvance` function is to check if the game event can advance the game state, i.e. it's well-formed and can be applied to the game.